### DISCUSSION TITLE
Filtering has many relationships with !=

### DISCUSSION THREAD

**Discussion by Hungnguyen720** [laravel/framework#35254] [Q&A] [answered]
Hello!

I have a query that filters posts based on category. It works when a post has only one category, the problem is when a post has more than one.

        return $query->whereHas('Categories', function ($q) {
            $q->where('categories.name', '!=', 'Archive');
        });

It works when a post belongs only to an archive category.
If a post belongs to an archive category and important category, I would like to not return the post.

Any help is much appreciated, Thank you!

SOLVED: Solution was to query relationship by absence using whereDoesntHave
https://laravel.com/docs/8.x/eloquent-relationships#querying-relationship-absence

**Comment by robert-moore96** [comment, marked as answer]
try using `->whereDoesntHave` instead 

https://laravel.com/docs/8.x/eloquent-relationships#querying-relationship-absence

**Comment by Hungnguyen720** [comment]
Thank you so much! You've saved me so much time, I went through the documentation for hours and missed it every time. 

My new query:

```
        return $query->whereDoesntHave('Categories', function ($q) {
            $q->where('name', '=', 'Archive');
        });
```
